ui/widgets/nuxeo-collapsible.js

```diff
@@ -0,0 +1,155 @@
+   *
+   * Example:
+   *
+   *     <nuxeo-collapsible expanded chevronPosition="start">
+   *        <label slot="heading">Nuxeo</label>
+   *        <div>Collapsed content</div>
+   *     </nuxeo-collapsible>
+   *
+   * ### Styling
+   * The following custom properties and mixins are available for styling:
+   * Custom property | Description | Default
+   * ----------------|-------------|----------
+   * `--nuxeo-collapsible-heading-padding` | Heading padding | `4px`
+   * `--nuxeo-collapsible-icon-height` | Chevron height | `20px`
+   * `--nuxeo-collapsible-icon-width` | Chevron width | `20px`
+   *
+   * @memberof Nuxeo
+   *
+   */
+
+  class Collapsible extends Nuxeo.Element {
+    static get template() {
+      return html`
+        <style>
+          .toggle {
+            --iron-icon-height: var(--nuxeo-collapsible-icon-height, 20px);
+            --iron-icon-width: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+
+          button {
+            height: 100%;
+            width: 100%;
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            margin: 0;
+            padding: var(--nuxeo-collapsible-heading-padding, 4px);
+            background: transparent;
+            border: none;
+            cursor: pointer;
+            text-decoration: none;
+            font-size: 1rem;
+          }
+
+          .heading {
+            width: calc(100% - var(--nuxeo-collapsible-icon-width, 20px));
+            overflow-x: hidden;
+            text-align: start;
+          }
+
+          .chevron {
+            height: var(--nuxeo-collapsible-icon-height, 20px);
+            width: var(--nuxeo-collapsible-icon-width, 20px);
+            display: block;
+          }
+
+          .start {
+            flex-direction: row-reverse;
+            justify-content: flex-end;
+          }
+
+          iron-collapse {
+            padding: 0 var(--nuxeo-collapsible-heading-padding, 4px);
+          }
+        </style>
+
+        <button aria-expanded="[[expanded]]" on-tap="_toggle" class$="[[chevronPosition]]">
+          <slot class="heading" name="heading"></slot>
+          <slot class="chevron" name="chevron">
+            <iron-icon class="toggle" icon="[[_toggleIcon(expanded)]]" toggle></iron-icon>
+          </slot>
+        </button>
+        <iron-collapse opened="[[expanded]]" no-animation="[[!animation]]">
+          <slot></slot>
+        </iron-collapse>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-collapsible';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * Set animation to false to disable animations
+         */
+        animation: {
+          type: Boolean,
+          value: true,
+        },
+
+        /**
+         * Chevron icon position on start or end of the button
+         */
+        chevronPosition: {
+          type: String,
+          value: 'end',
+        },
```

Comment:
as `chevronPosition` can takes only two values, I am wondering if we should  add validation about the given values .

Reply:
More than that, how the users/programmers know that it can only take two values?
I would suggest to have it as part of property documentation.

Reply:
Agreed. Changed the API slightly and now this prop is a boolean.